src/CognitoCustomMessageProcessor/Repository/ParticipantRegistrationDynamoDbRepository.cs

```diff
@@ -0,0 +1,49 @@
+using System.Collections.Generic;
+using System.Runtime.CompilerServices;
+using System.Threading;
+using System.Threading.Tasks;
+using Amazon.DynamoDBv2;
+using Amazon.DynamoDBv2.DataModel;
+using Amazon.DynamoDBv2.Model;
+using CognitoCustomMessageProcessor.Contracts;
+using CognitoCustomMessageProcessor.Settings;
+using Dte.Common.Persistence;
+
+namespace CognitoCustomMessageProcessor.Repository
+{
+    public class ParticipantRegistrationDynamoDbRepository : BaseDynamoDbRepository,
+        IParticipantRegistrationDynamoDbRepository
+    {
+        private readonly IAmazonDynamoDB _client;
+        private readonly DynamoDBOperationConfig _config;
+
+        public ParticipantRegistrationDynamoDbRepository(IAmazonDynamoDB client, IDynamoDBContext context,
+            AwsSettings awsSettings)
+            : base(client, context)
+        {
+            _client = client;
+            _config = new DynamoDBOperationConfig
+                { OverrideTableName = awsSettings.ParticipantRegistrationDynamoDbTableName };
+        }
+
+        public async Task<string> GetParticipantLocaleAsync(string participantId, [EnumeratorCancellation] CancellationToken cancellationToken = default)
```

Comment:
The `[EnumeratorCancellation]` attribute only applies to `IAsyncEnumerable` return types. You should remove it here.